bbc-basic/step7_quote.bbc

```diff
@@ -53,25 +53,29 @@ END
 DEF FNREAD(a$)
 =FNread_str(FNalloc_string(a$))
 
-DEF FNis_pair(val%)
-=FNis_seq(val%) AND NOT FNis_empty(val%)
-
-DEF FNquasiquote(ast%)
-  LOCAL car%, caar%
-  IF NOT FNis_pair(ast%) THEN =FNalloc_list2(FNalloc_symbol("quote"), ast%)
-  car% = FNfirst(ast%)
-  IF FNis_symbol(car%) THEN
-    IF FNunbox_symbol(car%) = "unquote" THEN =FNnth(ast%, 1)
-  ENDIF
-  IF FNis_pair(car%) THEN
-    caar% = FNfirst(car%)
-    IF FNis_symbol(caar%) THEN
-      IF FNunbox_symbol(caar%) = "splice-unquote" THEN
-        =FNalloc_list3(FNalloc_symbol("concat"), FNnth(car%, 1), FNquasiquote(FNrest(ast%)))
-      ENDIF
-    ENDIF
+DEF FNstarts_with(ast%, sym$)
+  LOCAL a0%
+  IF NOT FNis_seq(ast%) THEN =FALSE
+  a0% = FNfirst(ast%)
+  =FNis_symbol(a0%) AND FNunbox_symbol(a0%) = sym$
+
+REM  Map quasiquote, except to splice-unquoted elements.
+DEF FNqq_elts(seq%, env%)
+  LOCAL elt%, acc%
+  IF FNis_empty(seq%) THEN =seq%
+  elt% = FNfirst(seq%)
+  acc% = FNqq_elts(FNrest(seq%), env%)
+  IF starts_with(elt%, "splice-unquote") THEN
+    =FNcore_concat(FNEVAL(FNnth(elt%, 1), env%), acc%)
```

Comment:
`FNcore_concat` takes one argument, which is a list of all the sequences to concatenate, because BBC BASIC doesn't have variadic functions. You could assemble an argument list using `FNalloc_list2`, but it's simpler just to use `FNcore_concat1`:
```suggestion
    =FNcore_concat1(FNEVAL(FNnth(elt%, 1), env%), acc%)
```
Really, though, if `FNcore_concat1` is to be called from outside the `core` library it should be moved to `types` and renamed.

Reply:
Thanks. Please consider sending me a patch moving `more_concat1` to types, at least if you can test BBC-Basic locally.
I have created the branch `quasiquote-review.` in the `asarhaddon/mal` repository. I will merge it into the PR and trigger all tests again once each build failure has been investigated.